### PR TITLE
ports/minimal: Make this work for macOS

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: ports/minimal/Makefile [hunk 1]
```diff
@@ -26,9 +26,15 @@ CFLAGS_CORTEX_M4 = -mthumb -mtune=cortex-m4 -mcpu=cortex-m4 -msoft-float -fsingl
 CFLAGS += $(INC) -Wall -Werror -std=c99 -nostdlib $(CFLAGS_CORTEX_M4) $(COPT)
 LDFLAGS += -nostdlib -T stm32f405.ld -Map=$@.map --cref --gc-sections
 else
+UNAME := $(shell uname)
 LD = $(CC)
 CFLAGS += $(INC) -Wall -Werror -Wdouble-promotion -Wfloat-conversion -std=c99 $(COPT)
+ifeq ($(UNAME), Linux)
```
Comment (dpgeorge) [inline ports/minimal/Makefile:32]: Please remove the space before "Linux".

File: ports/minimal/Makefile [hunk 2]
```diff
@@ -26,9 +26,15 @@ CFLAGS_CORTEX_M4 = -mthumb -mtune=cortex-m4 -mcpu=cortex-m4 -msoft-float -fsingl
...
 LDFLAGS += -Wl,-Map=$@.map,--cref -Wl,--gc-sections
+else
+# assmue UNAME=Darwin
+LDFLAGS += -Wl,-map,$@.map
```
Comment (dpgeorge) [inline ports/minimal/Makefile:36]: I suggest adding `-Wl,-dead_strip` to these flags here.

File: ports/minimal/Makefile (hunk 1 above)
Comment (yamt) [reply ports/minimal/Makefile]: thank you. fixed.

i also made it similar to ports/unix and fixed the test.

**Comment by dpgeorge** [comment]
Thanks for the patch!

**Pull request merged**